Docs deploy

alongd opened the pull request:
This PR adds a GitHub Action for auto docs deployment using [github-pages-action](https://github.com/marketplace/actions/github-pages-action). A workflow file was added along with a requirements file, modifications were made to `.travis.yml`.
When reviewing, please carefully examine the changes made to `.travis.yml` to see that they are only docs-related modifications.

alongd committed:
Removed docs deployment code from .travis.yml

alongd committed:
TMP: Deploy the docs directly from this branch

alongd commented:
@bryanwweber, if you have a minute, please take a look at the error I get during deployment.
Any ideas how to resolve it? Am I missing a package?

![image](https://user-images.githubusercontent.com/16158262/103943566-fb3d8c80-513a-11eb-9ddd-43079299ae62.png)

bryanwweber commented:
Looks like you need to install Pandoc onto the virtual machine. You should be able to use apt in the Ubuntu images on GitHub actions to do that.

alongd committed:
Added requirements.txt

alongd committed:
install pandoc

bryanwweber commented on .github/workflows/gh-pages.yml:
```diff
@@ -0,0 +1,49 @@
+  deploy:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
```
I think you can do something like this:

```suggestion
      - run: sudo apt-get install pandoc
      - uses: actions/checkout@v2
```

alongd committed:
Added a GitHub Action to auto-deploy the docs

using "GitHub Pages action" (https://github.com/marketplace/actions/github-pages-action)

alongd committed:
f docs Makefile

alongd committed:
f gh

alongd committed:
f remove devtools

alongd committed:
TMP: test modifying the docs
